### PR TITLE
Added mount daemon process that is starts/stops w/ minikube start/stop

### PR DESCRIPTION
**Pull request by aaron-prindle**

### CODE REVIEW

File: cmd/minikube/cmd/mount.go [hunk 1]
```diff
@@ -46,6 +50,14 @@ var mountCmd = &cobra.Command{
 			fmt.Fprintln(os.Stderr, errText)
 			os.Exit(1)
 		}
+		if args[0] == "kill" {
```
Comment (r2d4) [inline cmd/minikube/cmd/mount.go]: I understand why you did this, but its a little weird to treat it as a subcommand without it actually using cobra.  It's probably not worth it to change the API, but adding a `mount add` and a `mount kill` would fix this (maybe even a `mount list`).

File: cmd/minikube/cmd/start.go [hunk 2]
```diff
@@ -194,8 +196,26 @@ func runStart(cmd *cobra.Command, args []string) {
...
+			cmdUtil.MaybeReportErrorAndExit(err)
+		}
+		err = cmdUtil.StoreProcessToFile(*mountCmd.Process, filepath.Join(constants.GetMinipath(), constants.MountProcessFileName))
+		if err != nil {
+			glog.Errorf("Error storing mount process information: %s", err)
+		}
+	}
 	if kubeCfgSetup.KeepContext {
-		fmt.Printf("The local Kubernetes cluster has started. The kubectl context has not been altered, kubectl will require \"--context=%s\" to use the local Kubernetes cluster.\n", kubeCfgSetup.ClusterName)
+		fmt.Printf("The local Kubernetes cluster has started. The kubectl context has not been altered, kubectl will require \"--context=%s\" to use the local Kubernetes cluster.\n",
```
Comment (r2d4) [inline cmd/minikube/cmd/start.go]: Should this have changed?

File: cmd/util/util.go [hunk 3]
```diff
@@ -210,3 +215,74 @@ minikube config set WantKubectlDownloadMsg false
...
+type SX map[string]interface{}
+
+func StoreProcessToFile(p os.Process, filename string) error {
+	encoded := ToGOB64(map[string]interface{}{
+		filepath.Base(filename): p,
+	})
+	return ioutil.WriteFile(filename, []byte(encoded), 0644)
+}
+
+func ReadProcessFromFile(filename string) (os.Process, error) {
```
Comment (r2d4) [inline cmd/util/util.go]: Any reason we can't just use `os.FindProcess(pid int)` here and only write the pid?

File: pkg/minikube/cluster/cluster_darwin.go [hunk 4]
```diff
@@ -66,8 +66,6 @@ func createXhyveHost(config MachineConfig) *xhyveDriver {
 		Boot2DockerURL: config.Downloader.GetISOFileURI(config.MinikubeISO),
 		BootCmd:        "loglevel=3 user=docker console=ttyS0 console=tty0 noembed nomodeset norestore waitusb=10 base host=" + cfg.GetMachineName(),
 		DiskSize:       int64(config.DiskSize),
-		Virtio9p:       true,
-		Virtio9pFolder: "/Users",
 		QCow2:          false,
```
Comment (r2d4) [inline pkg/minikube/cluster/cluster_darwin.go]: I think enabling `minikube mount` by default and adding the daemon should be separate PRs

File: pkg/minikube/constants/constants_linux.go [hunk 5]
```diff
@@ -18,7 +18,14 @@ limitations under the License.
...
+import (
+	"k8s.io/client-go/util/homedir"
+)
+
 var SupportedVMDrivers = [...]string{
 	"virtualbox",
 	"kvm",
 }
+
+var DefaultMountDir = homedir.HomeDir()
```
Comment (r2d4) [inline pkg/minikube/constants/constants_linux.go]: Maybe we can just switch this to `constants.DefaultMinipath`?

You probably only want to delete your mounts for the minikube profile you're working on.

File: pkg/minikube/constants/constants_windows.go [hunk 6]
```diff
@@ -18,7 +18,14 @@ limitations under the License.
...
+	"k8s.io/client-go/util/homedir"
+)
+
 var SupportedVMDrivers = [...]string{
 	"virtualbox",
 	"hyperv",
 }
+
+var DefaultMountDir = homedir.HomeDir()
+var DefaultMountEndpoint = "/hosthome"
```
Comment (r2d4) [inline pkg/minikube/constants/constants_windows.go]: Same here

File: cmd/minikube/cmd/start.go [hunk 7]
```diff
@@ -194,8 +196,26 @@ func runStart(cmd *cobra.Command, args []string) {
 		cmdUtil.MaybeReportErrorAndExit(err)
 	}
 
+	fmt.Printf("Setting up hostmount on %s:%s...\n", constants.DefaultMountDir,
+		constants.DefaultMountEndpoint)
+	// start 9p server mount
+	if !viper.GetBool(noMount) || cfg.GetMachineName() != constants.DefaultMachineName {
```
Comment (r2d4) [inline cmd/minikube/cmd/start.go]: Maybe we disable it by default to start off with?  

This is both adding the daemon to minikube mount and turning on mount by default.  Maybe split those into two PRs?  We should be able to test minikube mount daemon without turning it on by default right (with the mount integration test?)

File: cmd/util/util.go [hunk 8]
```diff
@@ -210,3 +215,74 @@ minikube config set WantKubectlDownloadMsg false
...
+	d := gob.NewDecoder(&b)
+	err = d.Decode(&m)
+	if err != nil {
+		fmt.Println(`failed gob Decode`, err)
+	}
+	return m
+}
+
+// Ask the kernel for a free open port that is ready to use
+func GetPort() (int, error) {
```
Comment (r2d4) [inline cmd/util/util.go]: Maybe if the only users of this expect a string, `GetPort() (string, error)` to avoid the conversions?

File: cmd/util/util.go (hunk 3 above)
Comment (aaron-prindle) [inline cmd/util/util.go]: At the time of writing this code I had thought that it would not be sufficient for windows to store the PID.  You could be correct though in that all that needs to be stored is the PID, I will test on Windows and add the result here.

File: pkg/minikube/cluster/cluster_darwin.go (hunk 4 above)
Comment (aaron-prindle) [inline pkg/minikube/cluster/cluster_darwin.go]: sounds good, will make this pr just enabling mount

File: pkg/minikube/constants/constants_linux.go (hunk 5 above)
Comment (aaron-prindle) [inline pkg/minikube/constants/constants_linux.go]: This is the dir that will be mounted into the VM.  So this will map the users home directory into the VM.  If this were constants.DefaultMinipath onlint $HOME/.minikube would be mounted.  I will keep this as for now

File: pkg/minikube/constants/constants_windows.go (hunk 6 above)
Comment (aaron-prindle) [inline pkg/minikube/constants/constants_windows.go]: (see above)

File: pkg/minikube/constants/constants_linux.go (hunk 5 above)
Comment (r2d4) [inline pkg/minikube/constants/constants_linux.go]: Ah makes sense. It's still a little weird to use the client-go import here. Can we use the std lib homedir?

File: cmd/minikube/cmd/start.go (hunk 2 above)
Comment (aaron-prindle) [inline cmd/minikube/cmd/start.go]: Added newline

File: cmd/util/util.go (hunk 8 above)
Comment (aaron-prindle) [inline cmd/util/util.go]: done

File: pkg/minikube/cluster/cluster_darwin.go (hunk 4 above)
Comment (aaron-prindle) [inline pkg/minikube/cluster/cluster_darwin.go]: done

File: pkg/minikube/constants/constants_linux.go (hunk 5 above)
Comment (r2d4) [inline pkg/minikube/constants/constants_linux.go]: Looked into it a little more.  Looks like `os/user` doesn't return the homedir for windows.  https://golang.org/src/os/user/lookup_windows.go#L127

I would just leave this as is for now then

File: cmd/util/util.go (hunk 3 above)
Comment (aaron-prindle) [inline cmd/util/util.go]: kill has a different implementation for process for windows and unix systems:
https://golang.org/src/os/exec_windows.go
https://golang.org/src/os/exec_unix.go

when you suggest kill by pid do you have an example of how that would be done cross platform?
it seems like it would be possible with just PID but these methods do use some additional information

**Comment by aaron-prindle** [comment]
PTAL @r2d4 all requested changes made

File: hack/jenkins/common.sh [hunk 9]
```diff
@@ -46,7 +46,7 @@ rm -rf $HOME/.minikube || true
 
 # Allow this to fail, we'll switch on the return code below.
 set +e
-out/e2e-${OS_ARCH} -minikube-args="--vm-driver=${VM_DRIVER} --v=10" -test.v -test.timeout=30m -binary=out/minikube-${OS_ARCH}
+out/e2e-${OS_ARCH} -minikube-args="--vm-driver=${VM_DRIVER} --v=10 --no-mount" -test.v -test.timeout=30m -binary=out/minikube-${OS_ARCH}
```
Comment (r2d4) [inline hack/jenkins/common.sh]: Lets run one set of tests with this on, then switch the no-mount default to true and rerun tests?  I think we should probably disable this by default

File: cmd/minikube/cmd/delete.go [hunk 10]
```diff
@@ -45,6 +50,21 @@ associated files.`,
...
 		}
 		fmt.Println("Machine deleted.")
+
+		out, err := ioutil.ReadFile(filepath.Join(constants.GetMinipath(), constants.MountProcessFileName))
+		if err != nil {
+			glog.Errorf("Error reading mount process from file: ", err)
+		}
+		pid, err := strconv.Atoi(string(out))
+		if err != nil {
+			glog.Errorf("Erro converting mount string to pid: ", err)
```
Comment (dlorenc) [inline cmd/minikube/cmd/delete.go]: typo: error

File: cmd/minikube/cmd/mount.go [hunk 11]
```diff
@@ -46,6 +50,23 @@ var mountCmd = &cobra.Command{
...
```
Comment (dlorenc) [inline cmd/minikube/cmd/mount.go]: Can this be deleted now that there's a kill command?

File: cmd/minikube/cmd/start.go (hunk 7 above)
Comment (dlorenc) [inline cmd/minikube/cmd/start.go]: +1

File: cmd/minikube/cmd/stop.go [hunk 12]
```diff
@@ -46,6 +52,21 @@ itself, leaving all files intact. The cluster can be started again with the "sta
 			cmdUtil.MaybeReportErrorAndExit(err)
 		}
 		fmt.Println("Machine stopped.")
+
+		out, err := ioutil.ReadFile(filepath.Join(constants.GetMinipath(), constants.MountProcessFileName))
```
Comment (dlorenc) [inline cmd/minikube/cmd/stop.go]: Can you refactor this and the delete implementation to a shared function?

File: cmd/minikube/cmd/mount.go (hunk 11 above)
Comment (aaron-prindle) [inline cmd/minikube/cmd/mount.go]: I am not sure what you are referring to.  This is so that a user who wants to stop the mount process that is spawned from `minikube start` has a way to do so without having to restart minikube.
```
minikube start
#don't want the mount
minikube mount kill
```

File: cmd/minikube/cmd/stop.go (hunk 12 above)
Comment (aaron-prindle) [inline cmd/minikube/cmd/stop.go]: done

**Comment by aaron-prindle** [comment]
requested changes made
